### DISCUSSION TITLE
WebRTC (beta 4, with build in Go2RTC) instead of Simple RTSP server

### DISCUSSION THREAD

**Discussion by Beer17HWAM** [fuatakgun/eufy_security#685] [Ideas]
>>>"2 Install RTSP Simple Server Add-on - Required for P2P Based Video Streaming - Not Required for RTSP Based Video Streaming"

I use this RTSP server (docker) explained in step 2 for my Eufy Doorbell (P2P).

Is it possible to use the WebRTC (beta 4, with build in Go2RTC) from AlexxIT (https://github.com/AlexxIT/WebRTC) ????

**Comment by fuatakgun** [comment]
some background on this topic;
I do want to ditch ffmpeg process that integration managing inside home assistant, just because, it is a pain for me. So I created a special rtsp_simple_server add-on and included ffmpeg inside it but it was taking 5 seconds to connect respective TCP port that integration is opening.
I also tried go2rtc, which includes ffmpeg naturally and it was taking more or less 5 seconds to connect tcp port.
I tried the initial solution with ffmpeg process, port and rtsp_simple_server and found out that, it is almost immediate, so I quickly ditched the initial idea.

This could be because of;
- my miss-configuration of docker network
- my inexperience with ffmpeg and video generation

anyway, so, until another cycle, I am not planning to revisit this idea, but, why are you interested with this?

**Reply by Beer17HWAM** [reply]
I have the Eufy Doorbell. I always used the thumbnail from the cloud for sending to my telegram bot. The fastest way to see what's happening at my front door. Now we only can start and stop the doorbell stream I am grabbing frames from the rtsp stream. I do this on my Raspberry pi with ffmpeg. I start the doorbell stream on motion and start a bash file which grabs frames and send those to Telegram. It works but a lot of the frames are distorted. I also tried the rtsp stream which passed WebRTC (go2rtc). I can grab frames from that stream too but also distortion. If I watch the eufy-security-ws stream directly in Home Assistant with WebRTC beta 4 it's fine. Also watching with a vlc player on a Windows system is displaying both streams fine, direct eufy-security-ws and via go2rtc. I try to find a quick way to grab frames after I start the Eufy stream. I am grabbing it myself directly on the Raspberry pi because Home Assistant needs time to see that the stream is available. If I setup a generic cam with the link to the Eufy rtsp stream. So maybe you can include a grab feature directly in the eufy-security-ws module.

**Reply by fuatakgun** [reply]
I think, you should just wait for encrypted image issue to be resolved and you can rely on cloud thumbnail then.

I included a service called generate_image under camera thubmnail on demand (normally it happens only when browser requested it), if camera is streaming.

**Reply by Beer17HWAM** [reply]
I know, but generate_image is not working. The cam icon is always gray.

**Comment by fuatakgun** [comment]
even though it is streaming?

**Reply by Beer17HWAM** [reply]
No, it's never working. Sometimes I can see the icon is grabbed from a stream because it's not a full gray image. But it's never a good picture. 

I am grabbing frames, every 2 seconds, with:

ffmpeg -rtsp_transport tcp -y -i rtsp://192.168.1.8:8554/T8210P0020330C10 -f image2 -vf fps=fps=1 /config/tmp/ffmpeg-snapshot.jpg

And send those to Telegram right away. 

Maybe you can include a command to grab a frame and save it to a file. Maybe directly from the cam.

**Reply by fuatakgun** [reply]
As I said earlier and as I am saying now, it is actually the same thing happening under generate_image here: https://github.com/fuatakgun/eufy_security/blob/43a3d0736b22a0d184af05028aa7a8a7219cafaa/custom_components/eufy_security/camera.py#L135

**Comment by Beer17HWAM** [comment]
I found a nice solution. I make a video and send it to Telegram. I record 4 seconds in very low resolution. I can see in Telegram that the first few frames are distorted, so I see a fraction of a second some gray frames but the rest of the video is perfect. Telegram is playing the video in the chat flow so it's great. I do this 40 seconds after the first motion detection. So I get a lot of small videos in my Telegram.

ffmpeg -y -i rtsp://192.168.1.8:8554/T8210P0020330C10 -s 320x240 -t 4 -vcodec libx264 /config/tmp/ffmpeg-snapshot.mp4

**Comment by Beer17HWAM** [comment]
ffmpeg -y -i rtsp://192.168.1.8:8554/T8210P0020330C10 -c copy -map 0 -t 6 /config/tmp/ffmpeg-snapshot.mp4

With -c copy -map 0 -t 6 it's even better. No distortion at all and 6 seconds of video. Pure a copy of the stream, no encoding.

**Comment by mikedrawback** [comment]
I'm interested in using go2rtc in place of rtsp simple server just because I'm running both addons and it would be nice to be able to run only one container for this purpose

**Reply by fuatakgun** [reply]
But how? Any suggestions?
